hollow/src/main/java/com/netflix/hollow/core/memory/encoding/HashCodes.java

```diff
@@ -168,13 +168,40 @@ public static int hashInt(int key) {
         return key;
     }
 
-    public static int hashTableSize(int numElements) {
+    /**
+     * Determine size of hash table capable of storing the specified number of elements with a load
+     * factor applied.
+     *
+     * @param numElements number of elements to be stored in the table
+     * @return size of hash table, always a power of 2
```

Comment:
@dkoszewnik is power-of-two a hard requirement or just a desirable trait?

Reply:
It's a hard requirement _unless_ you change some of the math.  See, for example:
https://github.com/Netflix/hollow/blob/master/hollow/src/main/java/com/netflix/hollow/core/read/engine/set/HollowSetTypeReadStateShard.java#L64

> `long bucket = startBucket + (hashCode & (endBucket - startBucket - 1))`

Here we're taking the mod of the hash code using (_the number of buckets_ - 1).  This bitmask only works because the number of buckets is a power of 2.

As a side note -- if you don't use a power of two, then the mod operation will be significantly more expensive.